components/brave_wallet/browser/brave_wallet_utils_unittest.cc

```diff
@@ -273,4 +273,284 @@ TEST(BraveWalletUtilsUnitTest, Mnemonic) {
   }
 }
 
+TEST(BraveWalletUtilsUnitTest, EncodeString) {
+  std::string output;
+  EncodeString("one", &output);
+  EXPECT_EQ(output,
+            // Count for input string.
+            "0x0000000000000000000000000000000000000000000000000000000000000003"
+            // Encoding for input string.
+            "6f6e650000000000000000000000000000000000000000000000000000000000");
+
+  output.clear();
+  EncodeString(
+      "oneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneoneon"
+      "e",
+      &output);
+  EXPECT_EQ(
+      output,
+      // Count for input string.
+      "0x0000000000000000000000000000000000000000000000000000000000000048"
+      // Encoding for input string.
+      "6f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e"
+      "656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f"
+      "6e65000000000000000000000000000000000000000000000000");
+
+  output.clear();
+  EncodeString("", &output);
+  EXPECT_EQ(
+      output,
+      "0x0000000000000000000000000000000000000000000000000000000000000000");
+}
```

Comment:
Would also be good to include some invalid input calls, I guess at least `!IsStringUTF8`